source/dub/dependency.d

```diff
@@ -34,6 +34,62 @@ struct PackageDependency {
 	Dependency spec;
 }
 
+struct GitDep{
+	private{
+		string urlPackage;
+		string commit;
+		string spec;
+	}
+
+	bool valid()const @safe{
+		return !!spec;
+	}
+
+	string toURL(){
+		return format(`%s/commit/%s`, urlPackage, commit);
+	}
+
+	URL toURLZip(){
+		URL ret=format(`%s/archive/%s.zip`, urlPackage, commit);
+		return ret;
+	}
+
+	string toVersion(){
+		auto ret=spec;
+		// IMPROVE
+		ret=ret.replace(`/`, `#`);
+		ret=ret.replace(`:`, `$`);
+		return ret;
+	}
+
+	bool isSame(ref const GitDep other)const @safe{
+		return this.commit==other.commit;
+	}
+
+	void from(string spec){
+		spec=spec.replace(`#`, `/`);
+		spec=spec.replace(`$`, `:`);
+		this.spec=spec;
+		/+
+		EG:
+		@https://github.com/CyberShadow/ae/commit/abc177c494c0ed75bc1671fbd4ebdd8323a4d734
+		+/
+		assert(spec[0]==Version.hashPrefix);
+		import std.regex;
+    auto m=spec[1..$].matchFirst(`^(.*?)/commit/([a-f0-9]+)$`.regex);
```

Comment:
this will compile the regex on every execution. While std.regex now comes with a bit of caching, it's still better to do sth. like 

```d
static re = `...`.regex;
spec[1..$].matchFirst(re);
````